R/rows-db.R

```diff
@@ -21,8 +24,10 @@
 #'   Set to `TRUE` to always check keys, or `FALSE` to never check.
 #'   The default is to check only if `in_place` is `TRUE` or `NULL`.
 #'
+#'   Currently these checks are no-ops and need yet to be implemented.
+#'
 #' @return A tbl object of the same structure as `x`.
-#'   If `in_place = TRUE`, [invisible] and identical to `x`.
+#'   If `in_place = TRUE`, a modified version of `x` is returned invisibly.
```

Comment:
`x` is guaranteed to be unchanged, only the contents of the underlying data. Same for the dm case.

Reply:
Could we please quickly compare our personal semantics here?

When we're talking about a `tbl object`, what do we mean? 
1. a reference to a table
2. the table itself (including everything)
3. the skeleton of a table (colnames, coltypes, data source)

I thought we were talking about 2. Does it depend on the data source what we mean by `tbl object`?

Reply:
I guess that locally a table including the data is returned and thus you need to say that `x` is modified (as it is done in rows.R). And on the other hand, the `x` referring to a remote table is just a reference to that table.
But then we can't really talk about the "same structure as `x`", when it's about a remote table, cause it would be rather meaningless, right?

Maybe as an alternative, we could write something like:

"A tbl object referring to a table with the same structure as `x`. If `in_place = TRUE`, [invisible] and referring to the now modified version of `x`."

Reply:
This is different for mutable and immutable data sources. The former only point to the data, the latter contain the data.

```suggestion
#'   If `in_place = TRUE`, the underlying data is updated as a side effect and `x` is returned, invisibly.
```